cmd/modelcmd/modelcommand.go

```diff
@@ -370,10 +369,9 @@ func (c *ModelCommandBase) modelFromStore(controllerName, modelIdentifier string
 		return "", nil, errors.Trace(err)
 	}
 
-	// If the identifier is 6-8 characters or a valid UUID,
+	// If the identifier is at least characters long,
 	// attempt to match one of the stored model UUIDs.
-	l := len(modelIdentifier)
-	if (l > 5 && l < 9) || names.IsValidModel(modelIdentifier) {
+	if len(modelIdentifier) > 6 {
```

Comment:
did you mean >6 (aka 7 or more) or did you intend >= 6?
Your comment also says "at least characters", I think you want a number in there.

Reply:
Sloppy :(

I have fixed this and added tests that verify it properly.